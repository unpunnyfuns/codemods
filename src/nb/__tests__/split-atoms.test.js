import { describe, expect, it } from 'vitest'
import { DEFAULT_TEST_OPTIONS, testTransform } from '../../__tests__/test-helper.js'

describe('nb/split-atoms', () => {
  it('splits barrel imports into individual atom imports', () => {
    const output = testTransform('nb/split-atoms', 'nb/split-atoms', 'tsx', {
      barrelImport: DEFAULT_TEST_OPTIONS.barrelImport,
      atomPrefix: DEFAULT_TEST_OPTIONS.atomPrefix,
    })
    expect(output).toMatchSnapshot()
  })
})
