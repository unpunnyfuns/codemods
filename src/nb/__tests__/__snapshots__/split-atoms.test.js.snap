// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nb/split-atoms > splits barrel imports into individual atom imports 1`] = `
"import { SomeOtherComponent } from '@org/common/src/other'

import { Box } from '@org/common/src/components/atoms/Box';
import { Button as Btn } from '@org/common/src/components/atoms/Button';
import { Typography } from '@org/common/src/components/atoms/Typography';
import type { ButtonProps } from '@org/common/src/components/atoms/Button';

export function MyComponent(_props: ButtonProps) {
  return (
    <Box>
      <Btn>Click me</Btn>
      <Typography>Hello</Typography>
      <SomeOtherComponent />
    </Box>
  )
}
"
`;
